### DISCUSSION TITLE
DI using UnityContainer

### DISCUSSION THREAD

**Discussion by dcosta-marex** [reqnroll/Reqnroll#337] [Q&A]
Hi all,

Wanted to ask if anyone has attempted to get ReqnRoll (or the latest version of Specflow) using UnityContainer as the Dependency Injection (DI) framework? And if so, how have you done it?

The context is the following:
- Currently upgrading an old but large solution from .NET Framework 4.6.x to .NET 8.0 that uses UnityContainer for DI
- There is a large number of the BDD tests running and using Specflow 2.x in .NET Framework with the runtime config using Unity, but this config was discontinued from version 3.x upwards according to the [release notes](https://docs.specflow.org/projects/specflow/en/latest/Installation/Breaking-Changes-with-SpecFlow-3.0.html#runtimeplugins-are-configured-by-references) from what we can see
- All Steps use DI to get our bespoke contexts, which in turn rely on what our libraries have registered
- We have been successful in getting Unity Container running in .NET 8.0 and we upgraded from Specflow to Reqnroll
- But after upgrading, when a test is executed, even though our libraries have registered all instances in the Unity Container, BoDI is not able to build and run the steps as it cannot find the required interfaces

Following the [docs](https://docs.reqnroll.net/latest/automation/context-injection.html#custom-dependency-injection-frameworks) recommendation, we are currently building a new runtime plugin using [this guide](https://docs.reqnroll.net/latest/extend/plugins.html#runtime-plugins) and the Autofac plugin as a template from github [here](https://github.com/reqnroll/Reqnroll/tree/main/Plugins/Reqnroll.Autofac.ReqnrollPlugin). Just wanted to confirm if anyone has been here before and has either build this plugin or has an alternative approach.

**Comment by gasparnagy** [comment]
The way how these plugins are built is a bit chaotic and we should make a simpler/clearer way to handle these. So whenever you play with your plugin, please keep this in mind and if you have any suggestion on how this could be done easier/better, please share it here.
